src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberCompleteWork
+ * @flow
+ */
+
+'use strict';
+
+import type { ReactCoroutine, ReactYield } from 'ReactCoroutine';
+import type { Fiber } from 'ReactFiber';
+
+var ReactChildFiber = require('ReactChildFiber');
+var ReactTypesOfWork = require('ReactTypesOfWork');
+var {
+  IndeterminateComponent,
+  FunctionalComponent,
+  ClassComponent,
+  HostComponent,
+  CoroutineComponent,
+  YieldComponent,
+} = ReactTypesOfWork;
+
+function transferOutput(unitOfWork : Fiber) {
+  var child = unitOfWork.child;
+  if (child && !child.sibling) {
+    // Common case
+    unitOfWork.output = child.output;
+    return;
+  }
+  var output = [];
+  while (child) {
+    output.push(child.output);
+    child = child.sibling;
+  }
+  unitOfWork.output = output;
+}
+
+function recursivelyFillYields(yields, output) {
```

Comment:
That is correct. I abandoned the "algebraic effects" model because I couldn't figure out how to make the cache invalidation work in all cases.

I think this is sufficient for layout, menu items etc. There can be a host component or other coroutines in the final tree due to continuations returning something to the outside, but they can't arbitrarily execute back and forth.

For example, this places a `<div />` between `<Foo />` and `<Bar />` in the tree.

```
const Foo = props => 
  createCoroutine(
    <Bar />,
    (props, yields) => <div>{yields[0]}</div>
  );
```